kiali/kiali
# Bookinfo gherkin annotations for upcoming hook

This is related to the OSSM-4049 ticket. 

In order to make more robust setup, we want Cypress to detect if certain demoapps are installed before related tests. For example, if the test uses bookinfo, then all of the 6 bookinfo pods and traffic generator must be installed. If they are not or if they are malformed at some way (for example missing pod), then Cypress installs the specific demo app via the hack scripts. 

To test this:
- Deploy Kiali without the bookinfo demo app
- Run any Cypres feature file, which has the `@bookino-app` tag.
- Bookinfo demo app should be installed prior to the first test and all tests related to bookinfo from that FeatureFile should pass
- This should work both for the OCP and non-OCP.

(Drafting this with bookinfo app only. I will turn it into a regular PR once I add other demos as well.
